### ISSUE TITLE
Proposal: Add linter for redundant :refer usage with namespace aliases

### ISSUE THREAD

**Issue by jramosg** [clj-kondo/clj-kondo#2734]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

[ To keep development of this project going, consider sponsoring. If you are
already a sponsor, thank you! You can remove this message. ]

**Is your feature request related to a problem? Please describe.**

Currently, clj-kondo does not warn when a var is both referred and accessed via a namespace alias in the same namespace. This can lead to inconsistent code style and confusion about which form to use. For example:

```clojure
(require '[clj-kondo.impl.utils :as utils :refer [sexpr]])
(utils/sexpr "")
(sexpr "")
```

Here, `sexpr` is both referred and accessed via the `utils` alias. This redundancy makes it unclear which style should be used and can clutter require statements.

**Describe the solution you'd like**

Add a linter (suggested name: `:aliased-referred-var`) that warns when a var is both referred and accessed via a namespace alias in the same namespace. The linter should not enforce a specific fix, but simply warn about the inconsistency so users can decide whether to:

- Remove the `:refer` and always use the namespaced version (e.g., `utils/sexpr`)
- Keep the `:refer` and use the unqualified version (e.g., `sexpr`) consistently

**Describe alternatives you've considered**

- Not warning about this at all (status quo)

[ Optional: if you are interested in doing a PR yourself, please leave a note. ]
I will try with a PR
